File: problem-solving/algorithms/time-conversion/time-conversion.js
function timeConversion(s) {
    const modifier = s.slice(-2);
    let [hh, mm, ss] = s.slice(0, -2).split(':').map(Number);

    if (hh === 12 && modifier === 'AM') {
        hh = 0;
    } else if (modifier === 'PM' && hh !== 12) {
        hh += 12;
    }

    hh = String(hh).padStart(2, '0');
    mm = String(mm).padStart(2, '0');
    ss = String(ss).padStart(2, '0');

    return `${hh}:${mm}:${ss}`;
}

File: problem-solving/algorithms/time-conversion/time_conversion.py
def timeConversion(s):
    time, modifier = s[:-2], s[-2:]
    hh, mm, ss = map(int, time.split(':'))

    if hh == 12 and modifier == 'AM':
        hh = 0
    elif modifier == 'PM' and hh != 12:
        hh += 12

    return f"{hh:02}:{mm:02}:{ss:02}"